### ISSUE TITLE
reading-in method comparison?

### ISSUE THREAD

**Issue by karabrug** [prickly-pythons/prickly-pythons#20]
re: the bonus tip from this week -- I’ve been using astropy.io’s ascii.read to import my .csv files. How does this compare to the pandas read_xxx?

**Comment by piyanatk** [comment]
@karabrug I am not familiar with astropy ascii module, but I think the answer to your question depends on your data. My recollection is that astropy ascii table is built to read astronomical data base table. pandas is more suitable for time-series data, but it only support up to 2D table data now. If you have general multidimensional data, [xarray](http://xarray.pydata.org/en/stable/) will work better.  (pandas depreciate 3D+ data support and redirect all 3D data uses to xarray. They are designed to work in conjunction.)

**Comment by jacluff1** [comment]
pandas.DataFrame only supports 2D arrays; but pandas.Series supports
whatever dimensional arrays you like.

Peace and long life,
Jacob Cluff
jacluff1@asu.edu
602-503-9202

**Comment by piyanatk** [comment]
Yes, you can store a flatten multidimensional array as a pandas Series, or store it as a nested Series, but if you use proper multidimensional coordinate references and selection, xarray will be better because it is designed specifically for that.

**Comment by chw3k5** [comment]
Like many things in python there is a huge number of way to read things. 

I like to use basic python (no external packages) to read in data. This way nothing is dependent on packages for when new people try to use my code and it works cross platform and on very small computers.  An example of that can be seen in my repository in [a file](https://github.com/chw3k5/fiskCubeSat/blob/master/chrisP/thermometerTest.py) that I made for an intern this summer. This can be the best if you data is mess and comes form multiple sources or has inconstant formatting, like strings and numbers, tab and comma, this way you can have control over every decision. This is also the hardest to learn.

You can also [read in using numpy](https://stackoverflow.com/questions/3518778/how-to-read-csv-into-record-array-in-numpy) if you want to work with floats or ints with you data.

If you do most of your work in pandas or astropy for processing you data then that might be the best choice (as long is it works for the file format you are reading). 

Next time you read something else in explore a new option and then collect them all.

**Comment by jacluff1** [comment]
I don't know about xarray, but I've used pandas.Series for 3D arrays and it
didn't flatten them. A Series lets you store and save whatever you want,
without having to same length or even same dimension arrays. You can store
floats, strings, dictionaries, 1D, 2D, ND arrays, and more all in one
Series.

Peace and long life,
Jacob Cluff
jacluff1@asu.edu
602-503-9202
